packages/typegpu/src/data/attributes.ts

```diff
@@ -0,0 +1,335 @@
+import {
+  type IMeasurer,
+  type IRefResolver,
+  type ISchema,
+  type ISerialInput,
+  type ISerialOutput,
+  MaxValue,
+  Measurer,
+  type Parsed,
+  type Unwrap,
+} from 'typed-binary';
+import { RecursiveDataTypeError } from '../errors';
+import {
+  type AnyTgpuData,
+  type AnyTgpuLooseData,
+  type ResolutionCtx,
+  type TgpuData,
+  type TgpuLooseData,
+  isDataLoose,
+  isDataNotLoose,
+} from '../types';
+import alignIO from './alignIO';
+
+// ----------
+// Public API
+// ----------
+
+export interface Align<T extends number> {
+  type: 'align';
+  alignment: T;
+}
+
+export interface Size<T extends number> {
+  type: 'size';
+  size: T;
+}
+
+export type AnyAttribute = Align<number> | Size<number>;
+
+export type AnyAttributeTuple = [...AnyAttribute[]];
```

Comment:
doesn't that evaluate to just `AnyAttribute[]`? what's the difference?

Reply:
I meant `type AnyAttributeTuple = AnyAttribute[];`
it looks to yield the same results, but I may be missing something

Reply:
or AnyAttribute[] directly instead of [...AnyAttribute[]]

Reply:
Good point, seems to work fine!